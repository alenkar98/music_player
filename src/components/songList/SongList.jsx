import SongRow from '../songRow/SongRow';
import './songList.css';
import { useSelector } from 'react-redux';

export default function SongList() {
  const songs = useSelector((state) => state.song.songs);
  const filteredSongs = useSelector((state) => state.song.filteredSongs);
  const queue = useSelector((state) => state.song.queue);

  return (
    <section>
      <div className="trackTable">
        <div className="list"></div>
        <div className="songName">Song name</div>
        <div className="artistName">Artist Name</div>
        <div className="track">Track</div>
        <div className="buttons"></div>
      </div>
      <div className="trackList">
        {queue.length > 1
          ? queue.map((song, index) => <SongRow key={index} {...song} />)
          : (filteredSongs.length > 0 ? filteredSongs : songs).map((song, index) => <SongRow key={index} {...song} />)}
      </div>
    </section>
  );
}
