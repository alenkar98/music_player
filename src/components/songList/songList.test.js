import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import SongList from './SongList';

const mockStore = configureStore([]);

test('SongList component renders correctly', () => {
  const initialState = {
    song: {
      songs: [],
      filteredSongs: [],
      queue: [],
    },
  };

  const store = mockStore(initialState);

  const component = renderer.create(
    <Provider store={store}>
      <SongList />
    </Provider>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
