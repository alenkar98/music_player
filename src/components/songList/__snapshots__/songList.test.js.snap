// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongList component renders correctly 1`] = `
<section>
  <div
    className="trackTable"
  >
    <div
      className="list"
    />
    <div
      className="songName"
    >
      Song name
    </div>
    <div
      className="artistName"
    >
      Artist Name
    </div>
    <div
      className="track"
    >
      Track
    </div>
    <div
      className="buttons"
    />
  </div>
  <div
    className="trackList"
  />
</section>
`;
